### PR TITLE
Refactor media handling in github_run.py

### PR DESCRIPTION
**Pull request by arash77**
This pull request updates the logic to handle media metadata, such as mentions and hashtags, and ensures that only relevant media is included in the metadata. Additionally, it updates the title of the pull request by using #.

### CODE REVIEW

File: github_run.py [hunk 1]
```diff
@@ -118,17 +118,19 @@ def new_pr(self, not_posted):
...
-                    for key, value in metadata.get("hashtags", {}).items()
-                    if key in media
-                }
-            new_md_content = f"---\n{yaml.dump(metadata)}---\n{text}"
+                if "mentions" in metadata:
+                    metadata["mentions"] = {
+                        key: value
+                        for key, value in metadata["mentions"].items()
+                        if key in media
+                    }
```
Comment (bgruening) [inline github_run.py:126]:
```suggestion
                    for key, value in metadata["mentions"].items():
                        if key in media:
                            metadata["mentions"].update({key: value})
```

or something like that, please ;)

Comment (arash77) [reply github_run.py]: but I think that might be different from what the other code is doing.
It is supposed to only keep the mentions and hashtags that are part of specific media.

File: github_run.py [hunk 2]
```diff
@@ -118,17 +118,14 @@ def new_pr(self, not_posted):
...
-                    if key in media
-                }
-                metadata["hashtags"] = {
-                    key: value
-                    for key, value in metadata.get("hashtags", {}).items()
-                    if key in media
-                }
-            new_md_content = f"---\n{yaml.dump(metadata)}---\n{text}"
+                for meta in ["mentions", "hashtags"]:
+                    if meta in metadata:
```
Comment (bgruening) [inline github_run.py:122]: this if is not needed I think you have `metadata.get(meta, {})` in place

Comment (arash77) [reply github_run.py:122]: This is to prevent the mentions and hashtags to create if they are empty.

File: github_run.py [hunk 3]
```diff
@@ -118,17 +118,14 @@ def new_pr(self, not_posted):
...
+                        new_metadata = {}
+                        for key, value in metadata.get(meta, {}).items():
```
Comment (bgruening) [inline github_run.py:124]:
```suggestion
                        for key, value in metadata[meta].items():
```

File: github_run.py (hunk 2 above)
Comment (bgruening) [reply github_run.py:122]: ok, then the get() is not needed, see below

**Review by bgruening — APPROVED** [review]
Thanks @arash77

**Pull request merged**